src/templates/head-meta.js

```diff
@@ -127,10 +127,30 @@ export function nuxtI18nHead ({ addDirAttribute = false, addSeoAttributes = fals
     const canonicalPath = currentRoute ? currentRoute.path : null
 
     if (canonicalPath) {
+      let href = toAbsoluteUrl(canonicalPath, baseUrl)
+
+      if (currentRoute && Array.isArray(canonicalQueries) && canonicalQueries.length) {
+        const currentRouteQueryParams = currentRoute.query
+        const params = new URLSearchParams()
+        for (const queryParamName of canonicalQueries) {
+          const queryParamValue = currentRouteQueryParams[queryParamName]
+          if (queryParamValue && Array.isArray(queryParamValue)) {
+            queryParamValue.forEach(v => params.append(queryParamName, v || ''))
+          }
+          if (queryParamValue && typeof queryParamValue === 'string') {
+            params.append(queryParamName, queryParamValue)
+          }
```

Comment:
Avoiding extra cycles.

```suggestion
          if (Array.isArray(queryParamValue)) {
            queryParamValue.forEach(v => params.append(queryParamName, v || ''))
          } else {
            params.append(queryParamName, queryParamValue)
          }
```

Reply:
@rchl I don't think this is equivalent. Doing an `if/else` means if there is no `queryParamValue` or the type is not a string, it's going to append it still.

Reply:
Why do you even check for type string?

Reply:
The type can only be a string or an array so it's redundant to check for the string type if already checked for array.

As for empty queryParamValue, maybe add a test for it. A query without the value (like `/?foo` should still be preserved.

Reply:
@rchl It is impossible to use `append` with URLSearchParams and pass no value, so we end up with a bunch of `?foo=undefined`